### PR TITLE
Refactor/new dapp boilerplate

### PR DESCRIPTION
**Pull request by stecb**
- New dapp boilerplate
- vite plugin pwa
- various enhancements/fixes

### CODE REVIEW

File: .env.template [hunk 1]
```diff
@@ -0,0 +1,30 @@
+SMART_CONTRACT_PATH=./contracts/deployments/networks.json
+
+# Variables injected in the HTML page
+
+HTML_TITLE=teledisko dao
```
Comment (vrde) [inline .env.template:5]:
```suggestion
HTML_TITLE=teledisko DAO
```

File: .env.template [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+# Open Graph Protocol properties. See https://ogp.me/
+HTML_DESCRIPTION=Teledisko dao
```
Comment (vrde) [inline .env.template:8]:
```suggestion
HTML_DESCRIPTION=teledisko DAO
```

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -2,15 +2,28 @@ name: Deploy
...
   deploy:
     runs-on: ubuntu-latest
 
+    env:
+      SMART_CONTRACT_PATH: ./contracts/deployments/networks.json
+      HTML_TITLE: Teledisko dao
+      HTML_DESCRIPTION: Teledisko dao
+      HTML_BASE_URL: https://dao.teledisko.com/
+      HTML_IMAGE: https://dao.teledisko.com/preview.png
+      HTML_IMAGE_ALT: Teledisko dao
```
Comment (vrde) [inline .github/workflows/deploy.yml:20]:
```suggestion
      HTML_IMAGE_ALT: teledisko DAO logo
```

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -2,15 +2,28 @@ name: Deploy
...
       - v*
 
 jobs:
   deploy:
     runs-on: ubuntu-latest
 
+    env:
+      SMART_CONTRACT_PATH: ./contracts/deployments/networks.json
+      HTML_TITLE: Teledisko dao
+      HTML_DESCRIPTION: Teledisko dao
```
Comment (vrde) [inline .github/workflows/deploy.yml:17]:
```suggestion
      HTML_DESCRIPTION: teledisko DAO
```

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -2,15 +2,28 @@ name: Deploy
...
+    tags:
       - v*
 
 jobs:
   deploy:
     runs-on: ubuntu-latest
 
+    env:
+      SMART_CONTRACT_PATH: ./contracts/deployments/networks.json
+      HTML_TITLE: Teledisko dao
```
Comment (vrde) [inline .github/workflows/deploy.yml:16]:
```suggestion
      HTML_TITLE: teledisko DAO
```

File: .env.template [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+HTML_BASE_URL=https://localhost:3000/
+HTML_IMAGE=https://localhost:3000/preview.png
+HTML_IMAGE_ALT=A hammer
+
+# Twitter properties. See:
+# https://developer.twitter.com/en/docs/twitter-for-websites/cards/overview/markup
+HTML_TWITTER_CARD=summary_large_image
+HTML_TWITTER_SITE=@vrde
```
Comment (vrde) [inline .env.template:16]: I guess we need to remove it or create a twitter handle, let's ignore it for nao

File: README.md [hunk 7]
```diff
@@ -1,13 +1,12 @@
-# teledisko dapp
+# Dapp template
```
Comment (vrde) [inline README.md:1]:
```suggestion
# Welcome to the teledisko DAO dapp
```

File: README.md [hunk 8]
```diff
@@ -1,13 +1,12 @@
...
 
-## How to develop
+This template should help get you started developing a dapp with Svelte and TypeScript in Vite.
```
Comment (vrde) [inline README.md:3]:
```suggestion
That's the code for the teledisko DAO dapp running on https://dao.teledisko.com/.
```

File: package.json [hunk 9]
```diff
@@ -1,92 +1,87 @@
 {
   "name": "teledisko-dao",
-  "version": "0.12.1",
+  "version": "0.20.0",
```
Comment (vrde) [inline package.json:3]: :rocket:

File: .env.template [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+# Variables injected in the HTML page
+
+HTML_TITLE=teledisko dao
+
+# Open Graph Protocol properties. See https://ogp.me/
+HTML_DESCRIPTION=Teledisko dao
+HTML_BASE_URL=https://localhost:3000/
+HTML_IMAGE=https://localhost:3000/preview.png
+HTML_IMAGE_ALT=A hammer
```
Comment (vrde) [inline .env.template:11]:
```suggestion
HTML_IMAGE_ALT=teledisko DAO logo
```

File: networks.json [hunk 11]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (vrde) [inline networks.json:1]: I guess we can remove this file no?

File: src/components/ReloadPrompt.svelte [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+
+  function close() {
+    needRefresh.set(false);
+  }
+</script>
+
+{#if $needRefresh}
+  <div class="pwa-toast" role="alert">
+    <Alert
+      message="New content available, click on Reload to update"
```
Comment (vrde) [inline src/components/ReloadPrompt.svelte:32]:
```suggestion
      message="A new version of the dapp is available. Would you like to update it now?"
```

File: src/components/ReloadPrompt.svelte [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+      type="info"
+    />
+    <Button variant="outlined" on:click={() => updateServiceWorker(true)}>
+      <Label>Reload</Label>
```
Comment (vrde) [inline src/components/ReloadPrompt.svelte:36]:
```suggestion
      <Label>Update</Label>
```

File: src/components/RuntimeErrors.svelte [hunk 14]
```diff
@@ -9,8 +9,10 @@
     {/each}
     <p>
       What to do now? Would be nice if you can
-      <a href="https://gitlab.com/teledisko/dao/-/issues/new" target="_blank"
-        >create an issue</a
+      <a
+        href="https://gitlab.com/teledisko/dao/-/issues/new"
```
Comment (vrde) [inline src/components/RuntimeErrors.svelte:13]:
```suggestion
        href="https://github.com/TelediskoDAO/dapp/issues/new"
```

File: src/stores/contracts.ts [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+import networks from "../../contracts/deployments/networks.json";
+
+import {
+  type Voting,
+  Voting__factory,
+  type TelediskoToken,
+  type ResolutionManager,
+} from "../../contracts/typechain";
+import { ResolutionManager__factory } from "../../contracts/typechain/factories/contracts/ResolutionManager/ResolutionManager__factory";
+import { TelediskoToken__factory } from "../../contracts/typechain/factories/contracts/TelediskoToken/TelediskoToken__factory";
```
Comment (vrde) [inline src/stores/contracts.ts:12]: You can also import them from here in theory `../../contracts/typechain`, no need to go deep. (Probably autocomplete failed to resolve these deps using the shortest path.)

**Review by vrde — APPROVED** [review]
I added few suggestions and comments. The sooner we merge the sooner we test it in staging :)

File: .env.template (hunk 10 above)
Comment (stecb) [reply .env.template:11]: LOOL (that's the template tho)

File: .env.template (hunk 6 above)
Comment (stecb) [reply .env.template:16]: if twitter will still exist... ⛵🪨

Comment (vrde) [reply .env.template:16]: FhN4NoVWQAMQjIq

**Pull request merged**